tensorflow/python/ops/rnn_cell_impl.py

```diff
@@ -1057,4 +1092,250 @@ def _linear(args,
+    custom_getter: Callable that allows overwriting the internal
+      get_variable method and has to have the same signature.
+    use_resource: If `True` use a ResourceVariable instead of a Variable.
+
+  Returns:
+    The created or existing variable.
+  """
+  collections = list(collections or [])
+  collections += [ops.GraphKeys.GLOBAL_VARIABLES, ops.GraphKeys.MODEL_VARIABLES]
+  var = variable(name, shape=shape, dtype=dtype,
+                 initializer=initializer, regularizer=regularizer,
+                 trainable=trainable, collections=collections,
+                 caching_device=caching_device, device=device,
+                 partitioner=partitioner, custom_getter=custom_getter,
+                 use_resource=use_resource)
+  return var
+
+def variable(name, shape=None, dtype=None, initializer=None,
+             regularizer=None, trainable=True, collections=None,
+             caching_device=None, device=None,
+             partitioner=None, custom_getter=None, use_resource=None):
+  """Gets an existing variable with these parameters or creates a new one.
+
+  Args:
+    name: the name of the new or existing variable.
+    shape: shape of the new or existing variable.
+    dtype: type of the new or existing variable (defaults to `DT_FLOAT`).
+    initializer: initializer for the variable if one is created.
+    regularizer: a (Tensor -> Tensor or None) function; the result of
+        applying it on a newly created variable will be added to the collection
+        GraphKeys.REGULARIZATION_LOSSES and can be used for regularization.
+    trainable: If `True` also add the variable to the graph collection
+      `GraphKeys.TRAINABLE_VARIABLES` (see `tf.Variable`).
+    collections: A list of collection names to which the Variable will be added.
+      If None it would default to `tf.GraphKeys.GLOBAL_VARIABLES`.
+    caching_device: Optional device string or function describing where the
+        Variable should be cached for reading.  Defaults to the Variable's
+        device.
+    device: Optional device to place the variable. It can be an string or a
+      function that is called to get the device for the variable.
+    partitioner: Optional callable that accepts a fully defined `TensorShape`
+      and dtype of the `Variable` to be created, and returns a list of
+      partitions for each axis (currently only one axis can be partitioned).
+    custom_getter: Callable that allows overwriting the internal
+      get_variable method and has to have the same signature.
+    use_resource: If `True` use a ResourceVariable instead of a Variable.
+
+  Returns:
+    The created or existing variable.
+  """
+  collections = list(collections if collections is not None
+                     else [ops.GraphKeys.GLOBAL_VARIABLES])
+
+  # Remove duplicates
+  collections = set(collections)
+  getter = vs.get_variable
+  with ops.device(device or ''):
+    return getter(name, shape=shape, dtype=dtype,
```

Comment:
this is a lot of boilerplate just to add a custom getter.  can you shorten this, remove the long docstring, and prepend a _ to the function name?  same for the other function.  i think most of the args can be passed as *args, **kwargs.